google-cloud-spanner-hibernate-testing/src/test/java/org/hibernate/userguide/schema/UniqueConstraintTest.java

```diff
@@ -55,17 +56,14 @@ public void test() {
             return author;
         } );
 
-        try {
+        // This should trigger a unique constraint violation.
+        assertThatThrownBy(() ->
             doInJPA( this::entityManagerFactory, entityManager -> {
-				Book book = new Book();
-				book.setTitle( "High-Performance Java Persistence" );
-				book.setAuthor( _author );
-				entityManager.persist( book );
-			} );
-        }
-        catch (Exception expected) {
-            assertNotNull( ExceptionUtil.findCause( expected, ConstraintViolationException.class ) );
-        }
+                Book book = new Book();
```

Comment:
Should we contribute this fix back to Hibernate repo?

Reply:
Sure, I can look into that.